### DISCUSSION TITLE
Unable to npm install exp branch in create-react-app project

### DISCUSSION THREAD

**Discussion by bsansone** [firebase/firebase-js-sdk#4397] [Q&A] [answered]
Attempted to install the experimental branch in my create-react-app project.

1. Cleared `node_modules`
2. Ran `npm i firebase@exp`
3. Error (screenshot attached)

My setup:
- MacBook Pro (Apple M1) Version 11.1
- Node v15.3
- Npm v7.0.14
- React v17.0.1
- React DOM v17.0.1
- React Scripts v4.0.0
- Firebase v9.3

<img width="832" alt="Screen Shot 2021-02-02 at 5 10 02 PM" src="https://user-images.githubusercontent.com/16868336/106679345-ad6d4600-6579-11eb-8898-6b585257fd30.png">

**Comment by hsubox76** [comment, marked as answer]
I didn't get any errors installing into a fresh create-react-app project.

You said you cleared node_modules, so it sounds like you're doing this in an existing project that previously had firebase installed. Did you also get rid of package-lock.json? Do you get the same error installing into a fresh create-react-app project?

**Reply by bsansone** [reply]
Removing `package-lock.json` did not fix the issue. Creating a new create-react-app project also failed on install. See screenshots:

<img width="832" alt="Screen Shot 2021-02-03 at 4 08 30 PM" src="https://user-images.githubusercontent.com/16868336/106821310-2502aa00-663a-11eb-9830-a3ce91d9a83b.png">

<img width="832" alt="Screen Shot 2021-02-03 at 4 08 32 PM" src="https://user-images.githubusercontent.com/16868336/106821349-3481f300-663a-11eb-8313-63bb89699d69.png">

**Reply by bsansone** [reply]
Seems like this is a false error? After this I ran `npm run start` in the fresh project and it ran fine, displaying the initial create-react-app screen on localhost

**Reply by hsubox76** [reply]
I can't reproduce these but I did identify a couple places that could possibly have caused both the database-types error and the installations-types error, so will look into it.

**Reply by hsubox76** [reply]
FYI, I was able to reproduce with your version of npm! (7.0.14) It doesn't seem to be a problem in the latest version of npm (7.5.2 as of now) - although that one has a bunch of warnings about something different.
